xbmc/GUILargeTextureManager.cpp

```diff
@@ -57,27 +57,33 @@ bool CImageLoader::DoWork()
   else
     loadPath = texturePath;
 
-  if (m_use_cache && loadPath.empty())
-  {
-    // not in our texture cache, so try and load directly and then cache the result
-    loadPath = CTextureCache::Get().CacheImage(texturePath, &m_texture);
-    if (m_texture)
-      return true; // we're done
-  }
   if (!m_use_cache || !loadPath.empty())
   {
```

Comment:
Right, looks indeed fishy. I'd remove loadpath from that check as `LoadFromFile` will return false anyways in case it's empty and we're told to not read from cache.